File: 프로그래머스/1/131128. 숫자 짝꿍/숫자 짝꿍.py
def solution(X,Y):
    X_numbers = [0] * 10
    Y_numbers = [0] * 10         
    answer = []
    
    for num in X:
        X_numbers[int(num)] += 1
    
    for num in Y:
        Y_numbers[int(num)] += 1

    for i in range(10):
        if int(X_numbers[i]) > 0 and int(Y_numbers[i]) > 0:
            for _ in range(min(X_numbers[i], Y_numbers[i])):
                answer.append(str(i))     
    
    answer.sort(reverse =True)
    
    if len(answer) == 0:
        return "-1"
    
    if answer[0] == "0":
        return "0"
    
    return "".join(answer)

File: 프로그래머스/1/131128. 숫자 짝꿍/숫자 짝꿍.js
function solution(X, Y) {
    
    var answer = [];
    
    X_dic = [...X].reduce((d,v) => {
        if (v in d) {
            d[v] += 1;
        }else {
           d[v] = 1; 
        }
        return d;
    },{});
    
    for(let c of [...Y]) {
        if (c in X_dic && X_dic[c] > 0) {
            X_dic[c] -= 1;
            answer.push(c);
        }  
    }
    if (answer.length != 0) {
        var result = answer.sort((a,b) => {
            if (a > b) {
                return -1
            }else {
                return 1
            }
        }).join("");
    }else {
        return "-1";
    }
    
    if (parseInt(result) == 0) {
        return "0";
    }
    
    return result;
}